Sources/JSONCheck/JSONCheck.swift

```diff
@@ -150,6 +150,74 @@ private func checkTestArrayTypes(context: JSONContext, _ testArray: [JSONValue])
     }
 }
 
+/// compare two `JSONValue`s and return an array of `CheckResult`s flagging all differences between the `test` and `reference` value that would make the statement `reference == test` false based on the conformance of `JSONValue` to `Equatable`.
+///
+/// - Parameters:
+///   - context: a `JSONContext` describing the location in a larger JSON structure where this check is taking place
+///   - reference: a reference `JSONValue` (the snapshot)
+///   - test: a `JSONValue` to be tested
+/// - Returns: an array of `CheckResult` diagnostics
+private func checkForStrictEquality(context: JSONContext, _ reference: JSONValue, _ test: JSONValue) -> [CheckResult] {
+    
+    switch (reference, test) {
+    case (.bool(let refBool), .bool(let testBool)):
+        guard refBool != testBool else { return .empty }
+        return [ .init(context: context, message: "Not a strict equality between the boolean \(refBool) and \(testBool)") ]
+        
+    case (.double(let refDouble), .double(let testDouble)):
+        guard refDouble != testDouble else { return .empty }
+        return [ .init(context: context, message: "Not a strict equality between the number \(refDouble) and \(testDouble)") ]
+        
+    case (.null, .null):
+        return .empty
+        
+    case (.string(let refString), .string(let testString)):
+        guard refString != testString else { return .empty }
+        return [ .init(context: context, message: "Not a strict equality between the string \"\(refString)\" and \"\(testString)\"") ]
+        
+    case (.object(let referenceObject), .object(let testObject)):
+        return checkForStrictEqualityOfObjects(context: context, referenceObject, testObject)
+        
+    case (.array(let referenceArray), .array(let testArray)):
+        guard referenceArray.count == testArray.count else {
+            return  [ .init(context: context, message: "Not a strict equality since arrays of different sizes: \(referenceArray.count) vs \(testArray.count)") ]
+        }
+        
+        return zip(referenceArray, testArray).enumerated().reduce(.empty) { previousChecks, acc in
```

Comment:
I think it would be better to not name the second parameter to this closure `acc` if possible — in the past when I’ve seen that used with reduce (and I like to use it too), `acc` stands for “accumulator”, which is traditionally the first argument to `reduce`, not the second. An alternate name for the variable could be something like `el` (for “element”) or `rec` (for “record”)!

Reply:
This is an awesome use of `zip` though!

Reply:
I like `el` !